airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MySqlSourceCdcTemporalConverter.kt

```diff
@@ -38,17 +39,65 @@ class MySqlSourceCdcTemporalConverter : RelationalColumnCustomConverter {
             TimestampHandler
         )
 
+    companion object {
+        // Logs column null-handling behavior.
+        fun logColumnNullHandling(
+            customConverterHandlerName: String,
+            column: RelationalColumn,
+            matches: Boolean
+        ) {
+            val log = KotlinLogging.logger {}
+            if (matches) {
+                if (column.isOptional) {
+                    log.info {
+                        "$customConverterHandlerName - Column '${column.name()}' is nullable, in case of zero-dates or NULL values, NULL will pass through"
+                    }
+                } else {
+                    log.warn {
+                        "$customConverterHandlerName - Column '${column.name()}' is NOT NULL, any NULL values (including zero-dates) will be converted to default epoch"
+                    }
+                }
+            }
+        }
+    }
+    /**
+     * Handling zero-dates (e.g., '0000-00-00 00:00:00'): Zero-dates arrive from Debezium in two
+     * forms:
+     * 1. As 0 (long)/epoch (1970-01-01...): typically for DEFAULT values or TIMESTAMP columns
+     * 2. As NULL: for actual zero-date data in rows, which is handled based on column nullability:
+     * ```
+     *    - NOT NULL columns: NULL is converted to epoch to prevent sync failures
+     *    - Nullable columns: NULL passes through unchanged
+     * ```
+     */
     data object DatetimeMillisHandler : RelationalColumnCustomConverter.Handler {
+        private var column: RelationalColumn? = null
 
-        override fun matches(column: RelationalColumn): Boolean =
-            column.typeName().equals("DATETIME", ignoreCase = true) &&
-                column.length().orElse(0) <= 3
+        override fun matches(column: RelationalColumn): Boolean {
+            this.column = column
```

Comment:
Is it always that a call to `matches()` is following by a call to `partialConverters`? 
If not there can be a mixup of different columns